httpbp/middlewares.go

```diff
@@ -158,3 +160,31 @@ func InjectEdgeRequestContext(truster HeaderTrustHandler, impl *edgecontext.Impl
 		}
 	}
 }
+
+// SupportedMethods returns a middleware that checks if the request is made
+// using one of the given HTTP methods.
+//
+// Returns a raw, plain text 405 error response if the method is not supported.
+func SupportedMethods(method string, additional ...string) Middleware {
+	supported := make(map[string]bool, len(additional)+1)
+	supported[strings.ToUpper(method)] = true
+	for _, m := range additional {
```

Comment:
🔕 Should we implicitly add HEAD as long as GET is allowed? Go's stdlib http package will handle HEAD requests automatically (drop the body and just return the headers you returned in your handler implementation), and developers are likely to forget add HEAD to the allowed list, and I think in most cases we don't really want to disallow HEAD requests.

Reply:
That makes sense